doc_src/cmds/path.rst

```diff
@@ -0,0 +1,359 @@
+.. _cmd-path:
+
+path - manipulate and check paths
+=================================
+
+Synopsis
+--------
+
+::
+
+    path base [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+    path dir  [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+    path extension [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+    path filter [(-z | --null-in)] [(-Z | --null-out)] [(-v | --invert)] [(-q | --quiet)] [(-t | --type) TYPE] [(-p | --perm) PERMISSION] [PATH...]
+    path expand [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [(-t | --type) TYPE] [(-p | --perm) PERMISSION] [GLOB...]
+    path match [(-z | --null-in)] [(-Z | --null-out)] [(-v | --invert)] [(-q | --quiet)] [(-t | --type) TYPE] [(-p | --perm) PERMISSION] GLOB [PATH...]
+    path normalize [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+    path real [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+    path strip-extension [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
+
+Description
+-----------
+
+``path`` performs operations on paths.
+
+PATH arguments are taken from the command line unless standard input is connected to a pipe or a file, in which case they are read from standard input, one PATH per line. It is an error to supply PATH arguments on the command line and on standard input.
+
+Arguments beginning with ``-`` are normally interpreted as switches; ``--`` causes the following arguments not to be treated as switches even if they begin with ``-``. Switches and required arguments are recognized only on the command line.
+
+All subcommands accept a ``-q`` or ``--quiet`` switch, which suppresses the usual output but exits with the documented status. In this case these commands will quit early, without reading all of the available input.
+
+All subcommands also accept a ``-z`` or ``--null-in`` switch, which makes them accept arguments from stdin separated with NULL-bytes. Since paths on uni can't contain NULL, that makes it possible to handle all possible paths and read input from e.g. ``find -print0``. If arguments are given on the commandline this has no effect.
+
+All subcommands also accept a ``-Z`` or ``--null-out`` switch, which makes them print output separated with NULL instead of newlines. This is for further processing, e.g. passing to another ``path`` with ``--null-in``, or ``xargs -0``. This is not recommended when the output goes to the terminal or a command substitution.
+
+Some subcommands operate on the paths as strings and so work on nonexistent paths, while others need to access the paths themselves and so filter out nonexistent paths.
+
+The following subcommands are available.
+
+.. _cmd-path-base:
+
+"base" subcommand
+--------------------
+
+::
+
+    path base [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [PATH...]
```

Comment:
Yeah, this is `base` purely because of `basename`. Ideally it'd be something like `filename`, which I believe people who aren't old UNIX-heads would actually look for, but that doesn't work because it might not be a "file" (except in the sense that a directory is also kind of a "file").

`last-component` feels like a bit of a mouthful, I'll have to think about that. Or does anyone have any better suggestions?

Reply:
Yeah I thought the same about "filename".
`path unqualify` is also too weird.
I'd say either stick to `path base` or go with something like `path strip-dir`, for symmetry with `path strip-extension`

Reply:
Let's keep base then, unless someone has a better idea.